flink/scripts/jobmanager-entrypoint.sh

```diff
@@ -0,0 +1,221 @@
+
+cd "$(dirname "${BASH_SOURCE[0]}")"
+source ./common.sh
+
+submit_job() {
+  log_warn "> Submiting job..."
+  ensure_flink_config
+  /opt/flink/bin/standalone-job.sh "$@" 1>&2 &
+  log_warn "> Job submited"
+}
+
+ensure_ha_directories() {
+  log_warn "> Ensuring directories existence..."
+  mkdir -p "$@"
+  log_warn "> Directories present"
+}
+
+get_savepoint_ref() {
+  local savepoint_ref_path savepoint_path
+  savepoint_ref_path="$1"
+  savepoint_path=''
+  if [ -f "$savepoint_ref_path" ]; then
+    log_warn "> Savepoint reference file found"
+    log_warn "> Geting savepoint path..."
+    savepoint_path=$(sed 's/#.*//g' < "$savepoint_ref_path")
+    log_warn "> Got savepoint: <$savepoint_path>"
+  else
+    log_warn "> Savepoint file not fount"
+  fi
+  echo "$savepoint_path"
+}
+
+clean_zookeeper_data() {
+  local zk_host zk_root_path job_name
+  zk_host="$1"
+  zk_root_path="$2"
+  job_name="$3"
+
+  local zk_clean_cmd
+  zk_clean_cmd="rmr $zk_root_path/$job_name"
+
+  local zk_cp_0 zk_cp_1 zk_cp_2 zk_cp
+  zk_cp_0="lib/org.apache.zookeeper.zookeeper-3.4.13.jar"
+  zk_cp_1="lib/org.slf4j.slf4j-api-1.7.26.jar"
+  zk_cp_2="org.apache.zookeeper.ZooKeeperMain"
+  zk_cp="$zk_cp_0:$zk_cp_1 $zk_cp_2"
+
+  log_warn "> Cleaning zookeeper path: <$zk_root_path/$job_name>"
+
+  echo "$zk_clean_cmd" | java -cp $zk_cp -server $zk_host
+
+  log_warn "> Zookeeper cleaned"
+}
+
+query_jobmanager_api_for_checkpoints() {
+  log_warn "> Querying rest api for checkpoints completed"
+  local rest_api_addr job_id api_response api_endpoint
+  rest_api_addr="$1"
+  # TODO get job id from rest api"
+  job_id="00000000000000000000000000000000"
+  api_endpoint="$rest_api_addr/jobs/$job_id/checkpoints"
+  log_warn "> Quering API in address: $api_endpoint"
+  api_response=$(curl -s "$api_endpoint"  || true)
+  log_warn "> Api response: <$api_response>"
+  echo "$api_response"
+}
+
+get_number_of_checkpoints_completed() {
+  log_warn "> Getting API response"
+  local api_response jq_path rest_api_addr
+  rest_api_addr="$1"
+  api_response=$(query_jobmanager_api_for_checkpoints "$rest_api_addr")
+  jq_path=".counts.completed"
+  num_checkpoint_completed=$(echo "$api_response" | jq "$jq_path")
+  log_warn "> Num of checkpoints completed: <$num_checkpoint_completed>"
+  echo "$num_checkpoint_completed"
+}
+
+ensure_checkpoint_completion() {
+  local rest_api_addr job_name timeout_in_secs interval_in_secs
+  rest_api_addr="$1"
+  job_name="$2"
+  timeout_in_secs="$3"
+  interval_in_secs="$4"
+
+  local current_time end_time
+  current_time=$(TZ=UTC0 date "+%s")
+  end_time=$((current_time + timeout_in_secs))
+
+  local num_checkpoint_completed
+  while [ "$current_time" -lt "$end_time" ]; do
+    current_time="$(TZ=UTC0 date '+%s')"
+    log_warn "> Verifying checkpoint creation..."
+
+    num_checkpoint_completed=$(get_number_of_checkpoints_completed \
+      "$rest_api_addr")
+    if [[ ${num_checkpoint_completed:+x} && $num_checkpoint_completed -gt 0 ]];
```

Comment:
What `${num_checkpoint_completed:+x}` does in this context? Would `num_checkpoint_completed` be always set in this context?

Reply:
I put it here to not crash the test if variable is empty. When api is not ready there is no response.

Reply:
Another option is to handle it in get_number_of_checkpoint_completed function

Reply:
Do you want me to implement this way?

Reply:
I'm thinking on just use `$num_checkpoint_completed` as-is. It won't break if it's empty or unset. Works for me:

```bash
#!/bin/bash
set -e
if [[ $num_checkpoint_completed -gt 0 ]]; then
        echo "gt than 0"
else
        echo "empty, not set, or bellow 0"
fi
```

```bash
$ ./test.sh
empty, not, set, or bellow 0
```